tests/components/yardian/test_config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Test the Yardian config flow."""
+from unittest.mock import AsyncMock, patch
+
+import pytest
+from pyyardian import NetworkException, NotAuthorizedException
+
+from homeassistant import config_entries
+from homeassistant.components.yardian.const import DOMAIN
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+pytestmark = pytest.mark.usefixtures("mock_setup_entry")
+
+
+async def test_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] == {}
+
+    with patch(
+        "homeassistant.components.yardian.config_flow.ConfigFlow.async_fetch_device_info",
```

Comment:
Patch the pypi code instead of patching home assistant here 

```suggestion
        "homeassistant.components.yardian.config_flow.AsyncYardianClient.fetch_device_info",
```